### DISCUSSION TITLE
Update states from single stream

### DISCUSSION THREAD

**Discussion by The-Mr-L** [rrousselGit/riverpod#1417] [Q&A]
Hi :) what is the best approach to update the state from  lets say websocket ,  I wonder if there is a way to declare the stream outside the ui context and then access the all the StateNotifierProviders  and  do the update based on stream event? like in getx this would just be done inside the GetController  inside the onInit method . 
I am pretty new to flutter and come from a backend dev world so sorry if this is a stupid question, thanks

**Comment by The-Mr-L** [comment]
okay I found out about ProviderContainer 
I stick with this for now :)

**Comment by rrousselGit** [comment]
As you maybe looking for StreamProvider?

**Reply by The-Mr-L** [reply]
well maybe? so how would I go about that, my use case is that I have a stream of bytes that comes from my backend in rust and then update all the  states from this stream based on enum value.  so it should be a ongoing stream in the background that keep updating the states?  is the streamProivder not meant to be used direct in the ui widget?

**Reply by The-Mr-L** [reply]
my setup atm is like so
I have a class that mange the stream and a provider I use to start it before the app inside the main fn using ProviderContainer 
```
final updateProvider = Provider<UpdateManager>((ref) {
  return UpdateManager(ref);
});


class UpdateManager{
  Ref ref;
  UpdateManager(this.ref);

  void init(){
    api.messageHandler().listen((event) {
      final msg = Payload.fromBuffer(event);
      switch (msg.kind) {
        case Payload_Kind.Balance:
          var balance = AccountBalance.fromBuffer(msg.payload);
          ref.read(accounts.notifier).update(balance.key, (a) => a.balance = balance.data);
          break;
       //etc
      }
    });
  }
}



```

is this a bad setup?
